juriscraper/pacer/appellate_attachment_page.py

```diff
@@ -0,0 +1,211 @@
+            "pacer_seq_no": self._get_pacer_seq_no(html_str),
+            "attachments": [],
+        }
+
+        for row in rows:
+            result["attachments"].append(
+                {
+                    "attachment_number": self._get_attachment_number(row),
+                    "description": self._get_description_from_tr(row),
+                    "page_count": self._get_page_count_from_tr(row),
+                    "pacer_doc_id": self._get_pacer_doc_id(row),
+                }
+            )
+        return result
+
+    def _get_dls_id(self):
+        """Extract the dls_id.
+
+        The dls_id is associated with the docket row - and is one number
+        off from the main document ID.  This should be used to identify
+        document number row.
+
+        In appellate attachment page, this is easy to extract with an XPath.
+        :return: dls_id
+        """
+        try:
+            rows = self.tree.xpath("//tbody/tr/td/a/@href")
+            pre_doc = sorted([row.split("/")[-1] for row in rows])[0]
+            return pre_doc[:3] + "0" + pre_doc[4:]
+        except IndexError:
+            return None
+
+    def _get_attachment_number(self, row):
+        """"Return the attachment number for an item.
+
+        :param row: Table row as an lxml element
+        :type: lxml HTML element
+        :return: Attachment number for row
+        :type: int
+        """
+        number = int(row.xpath(".//td/text()")[0].strip())
+        return number
+
+    def _get_description_from_tr(self, row):
+        """Get the description from the row
+
+        :param row: Table row
+        :type: lxml HTML element
+        :return: Row description
+        :type: unicode
+        """
+        description_text_nodes = row.xpath(".//td")
+        if not description_text_nodes:
+            return u""
+        return force_unicode(description_text_nodes[2].text_content().strip())
+
+    @staticmethod
+    def _get_page_count_from_tr(row):
+        """Take a row from the attachment table and return the page count as an
+        int extracted from the cell specified by index.
+        """
+        description_text_nodes = row.xpath(".//td")
+        if not description_text_nodes:
+            return None
+        return int(description_text_nodes[3].text_content().strip())
+
+    @staticmethod
+    def _get_pacer_doc_id(row):
+        """Take in a row from the attachment table and return the pacer_doc_id
+        for the item in that row. Return None if the ID cannot be found.
+        """
+        try:
+            row.xpath(u".//a")[0]
+        except IndexError:
+            # Item exists, but cannot download document. Perhaps it's sealed
+            # or otherwise unavailable in PACER. This is carried over from the
+            # docket report and may not be needed here, but it's a good
+            # precaution.
+            return None
+        else:
+            return row.xpath(".//td/a/@href")[0].split("/")[-1]
```

Comment:
Will you dig around a bit? I think we have code for this already. Maybe search for doc1?

Reply:
Fixed and updated